clients/drcachesim/tests/scheduler_unit_tests.cpp

```diff
@@ -367,58 +367,120 @@ test_regions()
+    // Test with synthetic streams and readers.
+    {
+        static constexpr memref_tid_t TID_A = 42;
+        static constexpr memref_tid_t TID_B = 99;
+        static constexpr memref_tid_t TID_C = 7;
+        std::vector<memref_t> refs_A = {
+            make_instr(TID_A, 50),
+            make_exit(TID_A),
+        };
+        std::vector<memref_t> refs_B = {
+            make_instr(TID_B, 60),
+            make_exit(TID_B),
+        };
+        std::vector<memref_t> refs_C = {
+            make_instr(TID_B, 60),
+            make_exit(TID_B),
+        };
+        std::vector<scheduler_t::input_reader_t> readers;
+        readers.emplace_back(std::unique_ptr<mock_reader_t>(new mock_reader_t(refs_A)),
+                             std::unique_ptr<mock_reader_t>(new mock_reader_t()), TID_A);
+        readers.emplace_back(std::unique_ptr<mock_reader_t>(new mock_reader_t(refs_B)),
+                             std::unique_ptr<mock_reader_t>(new mock_reader_t()), TID_B);
+        readers.emplace_back(std::unique_ptr<mock_reader_t>(new mock_reader_t(refs_C)),
+                             std::unique_ptr<mock_reader_t>(new mock_reader_t()), TID_C);
 
-    scheduler_t scheduler;
-    std::vector<scheduler_t::input_workload_t> sched_inputs;
-    sched_inputs.emplace_back(std::move(readers));
-    sched_inputs[0].only_threads.insert(TID_B);
-    if (scheduler.init(sched_inputs, 1,
-                       scheduler_t::make_scheduler_serial_options(/*verbosity=*/4)) !=
-        scheduler_t::STATUS_SUCCESS)
-        assert(false);
-    auto *stream = scheduler.get_stream(0);
-    memref_t memref;
-    uint64_t last_timestamp = 0;
-    for (scheduler_t::stream_status_t status = stream->next_record(memref);
-         status != scheduler_t::STATUS_EOF; status = stream->next_record(memref)) {
-        assert(status == scheduler_t::STATUS_OK);
-        assert(memref.instr.tid == TID_B);
+        scheduler_t scheduler;
+        std::vector<scheduler_t::input_workload_t> sched_inputs;
+        sched_inputs.emplace_back(std::move(readers));
+        sched_inputs[0].only_threads.insert(TID_B);
+        if (scheduler.init(sched_inputs, 1,
+                           scheduler_t::make_scheduler_serial_options(/*verbosity=*/4)) !=
+            scheduler_t::STATUS_SUCCESS)
+            assert(false);
+        auto *stream = scheduler.get_stream(0);
+        memref_t memref;
+        for (scheduler_t::stream_status_t status = stream->next_record(memref);
+             status != scheduler_t::STATUS_EOF; status = stream->next_record(memref)) {
+            assert(status == scheduler_t::STATUS_OK);
+            assert(memref.instr.tid == TID_B);
+        }
+    }
+    // Test with real files as that is a separate code path in the scheduler.
+    // Since 32-bit memref_t is a different size we limit these to 64-bit builds.
+#if (defined(X86_64) || defined(ARM_64)) && defined(HAS_ZLIB) && defined(HAS_SNAPPY)
+
+    {
```

Comment:
Just a nit, but why combine these rather than splitting this into two tests (synthetic streams and real files)?

Reply:
At first they were both going to test only_threads, but now the real-file test includes other things.  OK, split it into a separate test.